### ISSUE TITLE
[CoE Starter Kit - BUG] Core Components failed to import

### ISSUE THREAD

**Issue by sdadageico** [microsoft/coe-starter-kit#8026]
### Does this bug already exist in our backlog?

- [X] I have checked and confirm this is a new bug.

### Describe the issue

Solution "Center of Excellence - Core Components" failed to import: Solution Upgrade action failed after import as holding. InnerException is: Solution dependencies exist, cannot uninstall. DependencyCount : 5 RequiredComponent

### Expected Behavior

Solution imports successfully

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

4.24

### What app or flow are you having the issue with?

Core Components Solution

### What method are you using to get inventory and telemetry?

None

### Steps To Reproduce

login to make.powerapps.com
import solution COE core components
error appears as explained

### Anything else?

Solution "Center of Excellence - Core Components" failed to import: Solution Upgrade action failed after import as holding. InnerException is: Solution dependencies exist, cannot uninstall. DependencyCount : 5 RequiredComponentObject details: Type: connector, ObjectName: Office 365 Management API, Id: de4da0c8-bd02-41df-b0da-bbe0290e8fb4, ParentObjectName: , ParentId: 00000000-0000-0000-0000-000000000000, DependentComponentObject details: Type: connectionreference, ObjectName: admin_sharedadmin5foffice2036520management20api5f0fc7ec1df42e7af5_03750, Id: ff7bd356-84b4-ed11-83fe-000d3a1d5f48, ParentObjectName: , ParentId: 00000000-0000-0000-0000-000000000000 DependencyType: Published RequiredComponentObject details: Type: connector, ObjectName: Office 365 Management API, Id: de4da0c8-bd02-41df-b0da-bbe0290e8fb4, ParentObjectName: , ParentId: 00000000-0000-0000-0000-000000000000, DependentComponentObject details: Type: connectionreference, ObjectName: admin_sharedadmin5foffice2036520management20api5ff18d20b7ec3dbffe_a3e3b, Id: 05054dc5-6864-41e3-9499-490f0ac591e5, ParentObjectName: , ParentId: 00000000-0000-0000-0000-000000000000 DependencyType: Published RequiredComponentObject details: Type: connector, ObjectName: Office 365 Management API, Id: de4da0c8-bd02-41df-b0da-bbe0290e8fb4, ParentObjectName: , ParentId: 00000000-0000-0000-0000-000000000000, DependentComponentObject details: Type: connectionreference, ObjectName: admin_sharedadmin5foffice2036520management20api5f7fc9343cf45734fe_8f08c, Id: 8d632498-b647-434c-84ab-8eb00574102a, ParentObjectName: , ParentId: 00000000-0000-0000-0000-000000000000 DependencyType: Published RequiredComponentObject details: Type: connector, ObjectName: Office 365 Management API, Id: de4da0c8-bd02-41df-b0da-bbe0290e8fb4, ParentObjectName: , ParentId: 00000000-0000-0000-0000-000000000000, DependentComponentObject details: Type: Workflow, ObjectName: [Child] Admin | Sync Logs, Id: 751183fc-db68-ea11-a811-000d3a31cf04, ParentObjectName: , ParentId: 00000000-0000-0000-0000-000000000000 DependencyType: Published RequiredComponentObject details: Type: Attribute, ObjectName: Used by Apps (Number), Id: 43328aba-f4eb-4fe1-bdeb-b87f1cbc6108, ParentObjectName: PowerApps Connector, ParentId: 3e3da8bc-c0aa-4e6e-b1a4-9435e6c711fb, DependentComponentObject details: Type: SystemForm, ObjectName: Information, Id: f12456e0-69f3-4929-a868-ab020dd2a8e8, ParentObjectName: PowerApps Connector, ParentId: 00000000-0000-0000-

AB#2854

**Comment by sdadageico** [comment]
After I installed the solution there is an upgrade solution left in the environment but its the solution version that im trying to install

**Comment by ES00660463** [comment]
I have the same error. It's similar to #8024 

I'm trying to uninstall the upgrade version but only obtain errors.
![image](https://github.com/microsoft/coe-starter-kit/assets/85728639/b3866baa-bf53-4e69-bef8-ffa512c42f43)
This is the error when the process truying toquit the old solution:
![image](https://github.com/microsoft/coe-starter-kit/assets/85728639/d8cbf18f-6805-4c42-b68e-6e7f39cc2524)
Now, I've been trying to uninstall the update for half an hour and I can't do anything. Neither import, other solutions, nor delete either of the two core solutions.
![image](https://github.com/microsoft/coe-starter-kit/assets/85728639/57247fdd-6ca2-4224-8326-bd0d0b397e76)

**Comment by Jenefer-Monroe** [comment]
Hello, these are two different issues. @ES00660463 yours is an issue with teh product itself doing its upgrade. I'll hide your post above to avoid confusion. Please iterate on https://github.com/microsoft/coe-starter-kit/issues/8024 with notes as you go. 

OP's issue is a known issue from the deprecation of the Audit Logs solution back in August 2023. 

Here is the write up for how to resolve this: https://github.com/microsoft/coe-starter-kit/issues/6018
In short you will need to:

1. Delete _upgrade
2. Delete Audit Logs solution and any other dependencies
3. Try your upgrade of Core again
4. Setup new Audit Logs collection technique

**Comment by sdadageico** [comment]
How do I perform "Setup new Audit Logs collection technique"

**Comment by sdadageico** [comment]
Do you want me to re-install the audit log solutions I deleted ?

**Comment by Jenefer-Monroe** [comment]
Hello. Once you get through step 3 above you can setup Audit Logs again. Please see step 4 in https://github.com/microsoft/coe-starter-kit/issues/6018, there are two options (4a and 4b). 

![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/99387fe5-e724-499a-8c6d-56e2a2126c4b)

**Comment by sdadageico** [comment]
I was able to import the core solution thank you very much but I am facing another issue while running the COE Upgrade Wizard. The wizard skips the Configure Data Flows(Dat Export) step and directly moves to Share Apps

**Comment by Jenefer-Monroe** [comment]
Are you using Cloud Flow Architecture or Data Export? 

I presume the Cloud Flow (you haven't setup a data lake to ingest data from the product), and if so then this is by design, the Data Flows are only for the Data Export methodology.

**Comment by sdadageico** [comment]
The data export service is already setup though

**Comment by Jenefer-Monroe** [comment]
Ok if you are using Data Export then you dont need to collect audit logs. That data is natively gathered by the product. 

Have you ever set up using Data Export in the past? I ask as there is a big schema change coming and so if you havent done it before, you may be better served just waiting until that Feature Crew completes its work, and stick with the Cloud Flow for now. Up to you! 

If you want to use Data Export you need to select the option on this screen in the setup wizard, did you do that? 
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/9d373924-f514-47f3-b0ac-d39e3b057a8d)

**Comment by sdadageico** [comment]
The data export was already setup by my team but I will handle it going forward. I did not do the step mentioned above so I went back and doing it now. What do I need to do to accommodate the schema change ? Thank you very much for being pro-active

**Comment by AkshayKathuria06** [comment]
Hi Jenefer , Is it possible to setup audit log azure app with delegated permissions? My org isn't allowing application permission to be setup as it will grant access to all audit logs

**Comment by Jenefer-Monroe** [comment]
> The data export was already setup by my team but I will handle it going forward. I did not do the step mentioned above so I went back and doing it now. What do I need to do to accommodate the schema change ? Thank you very much for being pro-active

There is nothing you can do proactively there. Hopefully it works well for you as is as it does for many tenants. 
We will be working to respond to the new schema with the product team.

**Comment by Jenefer-Monroe** [comment]
> Hi Jenefer , Is it possible to setup audit log azure app with delegated permissions? My org isn't allowing application permission to be setup as it will grant access to all audit logs

If you are using Data Export methodology you dont need audit log at all. 

But if you do need audit logs, unfortunately you do need the application permissions to use the new http methodology. 
If you cannot get it, you can revert to using the old CC methodology but you will need to install these one off solutions we made for that, and then note that we have deprecated it and so wont be able to help with it. 

Please see 4b from above

> Hello. Once you get through step 3 above you can setup Audit Logs again. Please see step 4 in https://github.com/microsoft/coe-starter-kit/issues/6018, there are two options (4a and 4b). 
> 
> ![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/99387fe5-e724-499a-8c6d-56e2a2126c4b)
>

**Comment by AkshayKathuria06** [comment]
Thank you Jenefer.

**Comment by sdadageico** [comment]
I am not able to turn on the BYODL Flows as its asking for **admin_currentenvironment** Environment Variable value. What do I need to update here ?

**Comment by Jenefer-Monroe** [comment]
Are you using the setup wizard? You will not succeed at setup of the BYODL option without the setup wizard. It should set this for you.
